tegracli/utilities.py

```diff
@@ -0,0 +1,58 @@
+"""Utility functions for tegracli
+"""
+import datetime
+from functools import singledispatch
+from typing import Any, Dict, Union
+
+from telethon import TelegramClient
+
+from .types import AuthenticationHandler
+
+
+@singledispatch
```

Comment:
this is super fancy, but will be hard to read/maintain for most junior programmers/students we will hire in the future